source/RemoteWiring/RemoteDevice.cpp

```diff
@@ -159,6 +180,7 @@ RemoteDevice::pinMode(
 	uint8_t port_mask;
 	getPinMap( pin_, &port, &port_mask );
 
+    _firmata->lock();
```

Comment:
It was a great try, but you cannot use UWP objects in std::lock_guard object, because of the way the pointer is manipulated by the standard template library.